### PR TITLE
[rubysrc2cpg] Parser and ast changes for empty right hand side of association

### PR DESCRIPTION
**Pull request by karan-batavia**
@xavierpinho, can you please review?

### CODE REVIEW

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/parser/MethodDefinitionTests.scala [hunk 1]
```diff
@@ -604,4 +604,95 @@ class MethodDefinitionTests extends RubyParserAbstractTest {
     }
   }
 
+  "method definition for association parameters" should {
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/parser/MethodDefinitionTests.scala:607]: nitpick: these are called mandatory parameters

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 2]
```diff
@@ -889,6 +889,8 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
     "have correct structure for keyword usage inside association" in {
       val cpg = code("foo if: x.nil?")
 
+      println(cpg.call.name.l)
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:892]: should remove these

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 3]
```diff
@@ -1028,4 +1030,28 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
     literalArg.code shouldBe "/^ch/"
     literalArg.lineNumber shouldBe Some(3)
   }
+
+  "have correct structure when no RHS for an association as a parameter is provided" in {
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1034]: also mandatory parameters

**Review by xavierpinho — CHANGES_REQUESTED** [review]

**Comment by karan-batavia** [comment]
@xavierpinho , addressed your review.

File: joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala [hunk 4]
```diff
@@ -1054,4 +1054,30 @@ class SimpleAstCreationPassTest extends RubyCode2CpgFixture {
...
+    val cpg = code("""foo(bar:)""".stripMargin)
+
+    cpg.argument.size shouldBe 2
+    cpg.argument.l(0).code shouldBe "bar:"
+    cpg.argument.l(1).code shouldBe "bar"
+
+    cpg.call.size shouldBe 2
+    val List(callNode, operatorNode) = cpg.call.l
+    callNode.name shouldBe "foo"
+    operatorNode.name shouldBe "<operator>.activeRecordAssociation"
```
Comment (xavierpinho) [inline joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/SimpleAstCreationPassTest.scala:1068]: Out of this PR's responsibility, but might as well take advantage of a sample at hands.

I think there shouldn't be an operator here. The sample `foo(bar:)` should mean the same as `foo(bar: bar)`, in which case there wouldn't be an operator, but rather a call to `bar`.

Do you have a different opinion?

**Review by xavierpinho — APPROVED** [review]

**Comment by karan-batavia** [comment]
@xavierpinho, if there are no blockers, can you please merge?

**Comment by xavierpinho** [comment]
> @xavierpinho, if there are no blockers, can you please merge?

There are merge conflicts. Please rebase or merge with master and we should be good

**Comment by karan-batavia** [comment]
@xavierpinho Merged master.

**Pull request merged**